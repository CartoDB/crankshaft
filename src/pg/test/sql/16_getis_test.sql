\pset format unaligned
\set ECHO all
\i test/fixtures/getis_data.sql

-- set random seed
SELECT cdb_crankshaft._cdb_random_seeds(1234);

-- test against PySAL example dataset 'stl_hom'
SELECT rowid, round(z_score, 4) As z_score, round(p_value, 4) As p_value
FROM cdb_crankshaft.CDB_GetisOrdsG(
  'select * from getis_data',
  'hr8893', 'queen', NULL, 999,
  'the_geom', 'cartodb_id') As t(z_score, p_value, p_z_sim, rowid)
WHERE round(p_value, 4) <= 0.05
ORDER BY rowid ASC;
