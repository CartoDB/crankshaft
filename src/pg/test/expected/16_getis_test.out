\pset format unaligned
\set ECHO all
\i test/fixtures/getis_data.sql
SET client_min_messages TO WARNING;
\set ECHO none
_cdb_random_seeds

(1 row)
rowid|z_score|p_value
9|-0.7862|0.0500
22|-0.3955|0.0330
33|2.7045|0.0050
35|1.9524|0.0130
36|-1.2056|0.0170
37|3.4785|0.0020
38|-1.4622|0.0020
40|5.7098|0.0030
46|3.4704|0.0120
47|-0.9994|0.0320
48|-1.3650|0.0340
(11 rows)
